librad/src/profile.rs

```diff
@@ -37,51 +37,68 @@ impl Profile {
     /// the environment variables or well-known file.
     ///
-    /// If `RAD_PROFILE` is set, the value is used as the profile identifier
-    /// and system paths, scoped to the identifier, are used. See
-    /// [`Paths::new`] for details.
+    /// By default, the profile identifier is read from the `active_profile`
+    /// file in a system specific location. The paths returned by
+    /// [`Profile::paths`] are based on system paths (see [`directories::
+    /// ProjectDirs`]) and include the profile identifier. If the file
+    /// containing the identifier  does not exist, it is created and a new
+    /// identifier is generated and written to the file.
     ///
-    /// Otherwise and if the `RAD_HOME` environment variable is set, the value
-    /// is used to construct the profile with [`Profile::from_root`].
+    /// If the `RAD_PROFILE` environment variable is set, its value is used as
+    /// the profile identifier. The `active_profile` file is ignored.
     ///
-    /// Otherwise, the profile identifier is read from a well-known file and
-    /// system paths scoped to the identifier are used. If the file containing
-    /// the identifier  does not exist, it is created and a new identifier is
-    /// generated and written to the file.
+    /// If the `RAD_HOME` environment variable is set, its value is used instead
+    /// of system specific project directories. See also
+    /// [`Profile::from_root`].
     ///
     /// The profile identifier must not empty, must not contain path separators,
     /// must not be a windows path prefix like `C:`, and must not be a
-    /// special component like `.` or `..`.
+    /// special component like `.` or `..`. Otherwise an error is returned
     ///
     /// On Linux, the path to the active profile is
-    /// `$XDG_CONFIG_HOME/radicle-link/active_profile`.
+    /// `$XDG_CONFIG_HOME/radicle-link/active_profile` and profile specific
+    /// system paths are `$XDG_CONFIG_HOME/radicle-link/<profile_id>` and
+    /// `XDG_DATA_HOME/radicle-link/<profile-id>`.
     pub fn load() -> Result<Self, Error> {
-        if let Ok(id) = env::var("RAD_PROFILE") {
-            if !is_valid_profile_id(&id) {
-                return Err(Error::InvalidProfileIdFromEnv { id });
+        let env_profile_id = env::var("RAD_PROFILE").ok();
+
+        if let Some(ref id) = env_profile_id {
+            if !is_valid_profile_id(id) {
+                return Err(Error::InvalidProfileIdFromEnv { id: id.clone() });
             }
-            let paths = Paths::new(&id)?;
-            return Ok(Self { id, paths });
         }
 
         if let Ok(rad_home) = env::var("RAD_HOME") {
-            return Self::from_root(Path::new(&rad_home));
-        }
-
-        let id = load_profile_id(project_dirs()?.config_dir())?;
+            Self::from_root(Path::new(&rad_home), env_profile_id)
+        } else {
+            let id = if let Some(id) = env_profile_id {
```

Comment:
This suspiciously looks like `from_root` :eyes: Would I be right in thinking that this is `Self::from_root(project_dirs()?.config_dir()), env_profile_id)`? And then ultimately these just differ on the `root` being passed, i.e. `RAD_HOME` or the project dirs

Reply:
The difference between this and `Profile::from_root` is that here we call `Paths::new(id)` whereas in `Profile::from_root` we call `Paths::from_root(..)`; and both `Paths` constructors also differ. What this has in common with `Profile::from_root` is unwrapping the profile ID or loading it from file, but that’s not too much and doesn’t warrant extracting it.

Reply:
Ah you're right. Thanks for checking that :)